### ISSUE TITLE
Split PISM ocean models into "sub-shelf-surface" and "ocean proper"

### ISSUE THREAD

**Issue by ckhroulev** [pism/pism#310]
Similarly to *surface* and *atmosphere* models for the top boundary, PISM should have **shelf base** and **ocean** models (instead of one "ocean" lumping ocean forcing with parameterizations of sub-shelf melt).

**Rationale:** *atmosphere* models make it relatively easy to modify **inputs** of a top-surface-process model (using scalar offsets, for example). Currently *ocean modifiers* allow adjusting *outputs* of a selected ocean model, but not its inputs.

We would like to be able to use scalar offsets to modify the potential temperature of seawater used as an input of the `-ocean th` model, for example.

**Comment by tkleiner** [comment]
Is this still on the list? I think PISM would benefit.

**Comment by ckhroulev** [comment]
Yes, I still consider this worthwhile but we don't have the resources to work on this at the moment.

Please let me know if you (or somebody you know) would like to contribute the code. I'd be happy to help with such an effort.
